akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -454,6 +458,9 @@ private[akka] class RemoteActorRef private[akka] (
 
   @volatile var cachedAssociation: artery.Association = null
 
+  // used by artery to direct messages to a separate stream for large messages
+  @volatile var cachedLargeMessageDestinationFlag: LargeMessageDestinationFlag = null
+
```

Comment:
Yes, this is a trick that I considered for other things as well. It is fine to check-and-write on a volatile if the result is the same for all threads